src/main/java/duke/logic/api/requests/HttpRequest.java

```diff
@@ -0,0 +1,31 @@
+package duke.logic.api.requests;
+
+import duke.commons.exceptions.DukeApiException;
+
+/**
+ * Abstract class representing individual HTTP requests.
+ */
+public abstract class HttpRequest<T> {
```

Comment:
name sounds similar to some java class. do check and take care that you don't have name clashes with the java library.